src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java

```diff
@@ -934,6 +978,112 @@ public static CodonVariation createModification( final int codonId, final int co
         }
     }
 
+    // one or more codon variations that can be described together in standard nomenclature
+    @VisibleForTesting static final class CodonVariationGroup {
+        private final int[] refCodonValues;
+        private final StringBuilder altCalls;
+        private int startingCodon;
+        private int endingCodon;
+        private boolean isFrameShift;
+        private int insCount;
+        private int delCount;
+        private int subCount;
+
+        // start a new group with a first variation
+        public CodonVariationGroup( final int[] refCodonValues, final CodonVariation codonVariation ) {
+            this.refCodonValues = refCodonValues;
+            altCalls = new StringBuilder();
+            switch ( codonVariation.getVariationType() ) {
+                case FRAMESHIFT:
+                    isFrameShift = true;
+                    break;
+                case INSERTION:
+                    insCount += 1;
+                    altCalls.append(codonTranslation.charAt(codonVariation.getCodonValue()));
+                    break;
+                case DELETION:
+                    delCount += 1;
+                    break;
+                case MODIFICATION:
+                    subCount += 1;
+                    altCalls.append(codonTranslation.charAt(codonVariation.getCodonValue()));
+                    break;
+            }
+            startingCodon = endingCodon = codonVariation.getCodonId();
+        }
+
+        // attempt to add more variations
+        // returns false if the presented variation cannot be added to the group
+        public boolean addVariation( final CodonVariation codonVariation ) {
+            final int codonId = codonVariation.getCodonId();
+            if ( codonId > endingCodon + 1 && !isFrameShift ) return false;
+            switch ( codonVariation.getVariationType() ) {
+                case FRAMESHIFT:
```

Comment:
I am confused by the asymmetry between the above ctor and this modifier.
The ctor allows creation of a group with codon variation that is a `FRAMESHIFT`.
But here regardless if the group was constructed by a `FRAMESHIFT`, it always returns false.
I must be not understanding something.
Can you elaborate?

Reply:
you can never add a FRAMESHIFT to a group of variations that is being built.  a FRAMESHIFT always causes a new group to be started.  (the calling code is looping over variants, asking whether or not they can be incorporated into a growing group of variants that can be described by a single bit of standard nomenclature.  FRAMESHIFTs always stand alone.)

Reply:
And FRAMESHIFTs are unique.  you can only ever have one.

Reply:
I see. Thanks for explaining.